extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/handler/AstroThingHandler.java

```diff
@@ -164,112 +187,121 @@ private void restartJobs() {
         if (schedulerFuture != null && !schedulerFuture.isCancelled()) {
             schedulerFuture.cancel(true);
         }
-
-        schedulerFuture = scheduler.schedule(new Runnable() {
-            @Override
-            public void run() {
+        if (localExecutor != null) {
+            localExecutor.shutdownNow();
+        }
+        localExecutor = Executors.newSingleThreadExecutor();
```

Comment:
The `scheduler` is available throughout the lifetime of the handler. Here in this scenario, a lightweight single thread executor is introduced to submit only the immediate jobs that are executed when submitted and as soon as restart is requested, this local single thread executor is shutdown (forcefully to terminate the ongoing immediate jobs) and recreated. On the other hand, in this handler, scheduler is primarily used to schedule all the jobs for a time bound execution. We have one Scheduled Executor to schedule all the different jobs every 2ms and one single thread executor to invoke 2 jobs immediately.